File: src/main/java/com/seedsanskriti/service/AdminService.java
package com.seedsanskriti.service;

import java.util.List;

import com.seedsanskriti.dto.DashboardStatsResponse;
import com.seedsanskriti.dto.DeliveryResponse;
import com.seedsanskriti.dto.OrderResponse;
import com.seedsanskriti.dto.PaymentResponse;
import com.seedsanskriti.dto.ProductResponse;
import com.seedsanskriti.dto.SupplierResponse;
import com.seedsanskriti.dto.UpdateOrderStatusRequest;
import com.seedsanskriti.dto.UpdateSupplierStatusRequest;
import com.seedsanskriti.dto.UpdateUserStatusRequest;
import com.seedsanskriti.dto.UserResponse;

public interface AdminService {

    // ==========================
    // User Management
    // ==========================

    List<UserResponse> getAllUsers();

    UserResponse getUserById(Long userId);

    String updateUserStatus(UpdateUserStatusRequest request);

    // ==========================
    // Supplier Management
    // ==========================

    List<SupplierResponse> getAllSuppliers();

    SupplierResponse getSupplierById(Long supplierId);

    String updateSupplierStatus(UpdateSupplierStatusRequest request);

    // ==========================
    // Product Management
    // ==========================

    List<ProductResponse> getAllProducts();

    String deleteProduct(Long productId);

    // ==========================
    // Order Management
    // ==========================

    List<OrderResponse> getAllOrders();

    OrderResponse getOrderById(Long orderId);

    String updateOrderStatus(UpdateOrderStatusRequest request);

    // ==========================
    // Payment Management
    // ==========================

    List<PaymentResponse> getAllPayments();

    PaymentResponse getPaymentById(Long paymentId);

    // ==========================
    // Delivery Management
    // ==========================

    List<DeliveryResponse> getAllDeliveries();

    DeliveryResponse getDeliveryById(Long deliveryId);

    // ==========================
    // Dashboard
    // ==========================

    DashboardStatsResponse getDashboardStats();

}

File: seedsanskriti_frontend/src/services/adminService.js
import apiClient from './apiClient';

const adminService = {
  // Users
  getAllUsers: () => apiClient.get('/admin/users').then((res) => res.data),
  getUserById: (userId) => apiClient.get(`/admin/users/${userId}`).then((res) => res.data),
  updateUserStatus: (payload) =>
    apiClient.put('/admin/users/status', payload).then((res) => res.data),

  // Suppliers
  getAllSuppliers: () => apiClient.get('/admin/suppliers').then((res) => res.data),
  getSupplierById: (supplierId) =>
    apiClient.get(`/admin/suppliers/${supplierId}`).then((res) => res.data),
  updateSupplierStatus: (payload) =>
    apiClient.put('/admin/suppliers/status', payload).then((res) => res.data),

  // Products
  getAllProducts: () => apiClient.get('/admin/products').then((res) => res.data),
  deleteProduct: (productId) =>
    apiClient.delete(`/admin/products/${productId}`).then((res) => res.data),

  // Orders
  getAllOrders: () => apiClient.get('/admin/orders').then((res) => res.data),
  getOrderById: (orderId) => apiClient.get(`/admin/orders/${orderId}`).then((res) => res.data),
  updateOrderStatus: (payload) =>
    apiClient.put('/admin/orders/status', payload).then((res) => res.data),

  // Payments
  getAllPayments: () => apiClient.get('/admin/payments').then((res) => res.data),
  getPaymentById: (paymentId) =>
    apiClient.get(`/admin/payments/${paymentId}`).then((res) => res.data),

  // Deliveries
  getAllDeliveries: () => apiClient.get('/admin/deliveries').then((res) => res.data),
  getDeliveryById: (deliveryId) =>
    apiClient.get(`/admin/deliveries/${deliveryId}`).then((res) => res.data),

  // Dashboard
  getDashboardStats: () => apiClient.get('/admin/dashboard').then((res) => res.data),
};

export default adminService;
